cmd/doctor.go

```diff
@@ -495,3 +501,87 @@ func runDoctorScriptType(ctx *cli.Context) ([]string, error) {
 	}
 	return []string{fmt.Sprintf("ScriptType %s is on the current PATH at %s", setting.ScriptType, path)}, nil
 }
+
+func runDoctorCheckDBConsistency(ctx *cli.Context) ([]string, error) {
+	_, committer, err := models.TxDBContext()
+	if err != nil {
+		return nil, err
+	}
+	sess := committer.(models.Engine)
+	defer committer.Close()
+	var results []string
+
+	//find issues without existing repository
+	count, err := sess.Table("issue").
+		Join("LEFT", "repository", "issue.repo_id=repository.id").
+		Where("repository.id is NULL").
+		Count("id")
+	if err != nil {
+		return nil, err
+	}
+	if count > 0 {
+		if ctx.Bool("fix") {
+			if _, err = sess.In("id", builder.Select("issue.id").
+				From("issue").
+				Join("LEFT", "repository", "issue.repo_id=repository.id").
+				Where(builder.IsNull{"repository.id"})).
+				Delete(models.Issue{}); err != nil {
```

Comment:
We'd need to delete all the associated records as well (comments, watches, etc.) I'd consider calling the `models` package delete function instead, if it can be adapted for this.

Reply:
on it ...

Reply:
will be hard to backport